WinUIGallery/App.xaml.cs

```diff
@@ -1,288 +1,273 @@
+        protected override void OnLaunched(Microsoft.UI.Xaml.LaunchActivatedEventArgs args)
+        {
+            IdleSynchronizer.Init();
+
+            startupWindow = WindowHelper.CreateWindow();
+            startupWindow.ExtendsContentIntoTitleBar = true;
+#if DEBUG
+            //if (System.Diagnostics.Debugger.IsAttached)
+            //{
+            //    this.DebugSettings.EnableFrameRateCounter = true;
+            //}
+
+            if (System.Diagnostics.Debugger.IsAttached)
+            {
+                this.DebugSettings.BindingFailed += DebugSettings_BindingFailed;
+            }
+#endif
+
+            EnsureWindow();
+        }
+
+        private void DebugSettings_BindingFailed(object sender, BindingFailedEventArgs e)
+        {
+
+        }
+
+#if WINUI_PRERELEASE
+        protected override void OnActivated(IActivatedEventArgs args)
+        {
+            EnsureWindow(args);
+        }
+#endif
+
+        private async void EnsureWindow(IActivatedEventArgs args = null)
+        {
+            // No matter what our destination is, we're going to need control data loaded - let's knock that out now.
+            // We'll never need to do this again.
+            await ControlInfoDataSource.Instance.GetGroupsAsync();
+            await IconsDataSource.Instance.LoadIcons();
+
+            Frame rootFrame = GetRootFrame();
+
+            ThemeHelper.Initialize();
+
+            Type targetPageType = typeof(NewControlsPage);
+            string targetPageArguments = string.Empty;
+
+            if (args != null)
+            {
+                if (args.Kind == ActivationKind.Launch)
+                {
+                    if (args.PreviousExecutionState == ApplicationExecutionState.Terminated)
+                    {
+                        try
+                        {
+                            await SuspensionManager.RestoreAsync();
+                        }
+                        catch (SuspensionManagerException)
+                        {
+                            //Something went wrong restoring state.
+                            //Assume there is no state and continue
+                        }
+                    }
+
+                    targetPageArguments = ((Windows.ApplicationModel.Activation.LaunchActivatedEventArgs)args).Arguments;
+                }
+            }
+            var eventargs = Microsoft.Windows.AppLifecycle.AppInstance.GetCurrent().GetActivatedEventArgs();
+            if (eventargs != null && eventargs.Kind is ExtendedActivationKind.Protocol && eventargs.Data is ProtocolActivatedEventArgs)
+            {
+                ProtocolActivatedEventArgs ProtocolArgs = eventargs.Data as ProtocolActivatedEventArgs;
+                var uri = ProtocolArgs.Uri.LocalPath.Replace("/", "");
+
+                targetPageArguments = uri;
+                string targetId = string.Empty;
+                File.WriteAllText("D:\\uri.txt", uri);

```

Comment:
Debug code :)? This crashes the app on my machine as it's not able to find this file